OneDrive-Cloud-Player/App.xaml.cs

```diff
@@ -34,11 +36,39 @@ public App()
         {
             Core.Initialize();
             this.InitializeComponent();
+            this.LoadUserSettings();
             this.CreateScopedPublicClientApplicationInstance();
             this.Suspending += Application_Suspending;
             this.CacheHelper = new CacheHelper();
         }
 
+        /// <summary>
+        /// Load the user settings from disk and check if they contain all the required entries.
+        /// </summary>
+        private void LoadUserSettings()
+        {
+            // Load the saved settings from disk
+            UserSettings = ApplicationData.Current.LocalSettings;
+
+            // Create a dictionary with the required default settings
+            Dictionary<string, object> defaultSettings = new Dictionary<string, object>
+            {
+                { "MediaVolume", 100 },
+                { "ShowDefaultSubtitles", true }
+            };
+
+            // Check if the settings on disk contain all the required entries.
+            // If not, add them with a default value.
+            // TODO: instead of looping through the default list, loop through list from disk to remove obsolete settings
```

Comment:
What do you mean exactly by "Looping through a list to remove obsolete settings"?

Reply:
Say, for instance, we add a whole bunch of settings. These are then written to disk. As it is right now, that settings file on disk can only grow, not shrink. If we remove any entries from the program in the future, these will stay on everyone's disk untill the end of time. Admittedly, this is not a large problem for the settings file, but an inconsistency nonetheless.

Reply:
I personally don't think it is a big problem to have some used bytes on the disk. But if we really want to dive into this in the future we can detect the app version on startup to see if it has changed. If it had changed it will run a "cleaning" process and / or other processes that would be handy or needed after an update.

Reply:
Added in latest commit.